### PR TITLE
 UI: ConfirmButton component

### PR DESCRIPTION
**Pull request by alexanderzobnin**
This PR adds new `ConfirmButton` component. Also, new `link` variant added for `Button`.

### CODE REVIEW

**Review by torkelo — COMMENTED** [review]
Can this replace the DeleteButton? they seem too similar to have both

**Comment by alexanderzobnin** [comment]
Yes, sure, that was tricky in a big PR, but now it's a good chance to do it.

File: packages/grafana-ui/src/components/Button/Button.story.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ const defaultProps = {
 
 const variants = {
   size: ['xs', 'sm', 'md', 'lg'],
-  variant: ['primary', 'secondary', 'danger', 'inverse', 'transparent'],
+  variant: ['primary', 'secondary', 'danger', 'inverse', 'transparent', 'link'],
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Button/Button.story.tsx]: link is not a variant but different component (Button & LinkButton)

**Review by torkelo — CHANGES_REQUESTED** [review]
Think this looks ok I think, but adding the link variant looks strange, maybe you missed the LinkButton? (they have different props, which is why they are different Components)

File: packages/grafana-ui/src/components/Button/Button.story.tsx (hunk 1 above)
Comment (alexanderzobnin) [reply packages/grafana-ui/src/components/Button/Button.story.tsx]: `LinkButton` has the same variants (styles) as `Button`. But we need a variant where button will look like link (blue underscored text). If I use `LinkButton` with `transparent` variant, it wouldn't look like a link:

Screenshot from 2019-12-10 17-55-56

Comment (alexanderzobnin) [reply packages/grafana-ui/src/components/Button/Button.story.tsx]: You said [here](https://github.com/grafana/grafana/pull/20498#discussion_r352539316) 
>This component should be removed we have a LinkButton component already, use the "transparent" variant and restyle it and rename it to "link"

Could you explain, maybe I misunderstood what you want?

Comment (dprokop) [reply packages/grafana-ui/src/components/Button/Button.story.tsx]: I agree with @torkelo here. Maybe we could have a `link` variant which is only available for LinkButton that renders what you want here? 

Also, we need to introduce this new variant to the new button components (Forms/Button). I think we could actually introduce it only for the new buttons, and use the new styles instead

**Review by dprokop — CHANGES_REQUESTED** [review]
@alexanderzobnin agree with @torkelo  about the link button variant. I also think we can improve the LinkButton component and make it render as a link by default ('link' variant as default).  Apart from that please see my comment here https://github.com/grafana/grafana/pull/20993/files#r356437851

File: packages/grafana-ui/src/components/Button/Button.story.tsx (hunk 1 above)
Comment (alexanderzobnin) [reply packages/grafana-ui/src/components/Button/Button.story.tsx]: @dprokop To be clear, do you want to add `link` variant to `LinkButton` only?

Comment (dprokop) [reply packages/grafana-ui/src/components/Button/Button.story.tsx]: I think this variant makes sense only in LinkButton

**Comment by torkelo** [comment]
Wait, @dprokop think we need to rethink this a bit, the question of if it should be a LinkButton or Button is not about looks, but what element it is (has it a href), and this case it's a button not a link really. 

Maybe this is a problem with the design?

File: public/app/features/api-keys/ApiKeysPage.tsx [hunk 2]
```diff
@@ -287,7 +287,14 @@ export class ApiKeysPage extends PureComponent<Props, any> {
                     <td>{key.role}</td>
                     <td>{this.formatDate(key.expiration)}</td>
                     <td>
-                      <DeleteButton onConfirm={() => this.onDeleteApiKey(key)} />
+                      <ConfirmButton
```
Comment (alexanderzobnin) [inline public/app/features/api-keys/ApiKeysPage.tsx]: @torkelo @dprokop I noticed after removing `DeleteButton` we use `ConfirmButton` with the same props in several places. Does it make sense to recreate `DeleteButton` based on new `ConfirmButton`? Maybe do not place it in `@grafana/ui`, but in core grafana as a reusable componen?

Comment (dprokop) [reply public/app/features/api-keys/ApiKeysPage.tsx]: @alexanderzobnin maybe instead of recreating it, simply replace it with confirm button?

Comment (alexanderzobnin) [reply public/app/features/api-keys/ApiKeysPage.tsx]: @dprokop Maybe. I just don't know what's a better place for this component.

Comment (dprokop) [reply public/app/features/api-keys/ApiKeysPage.tsx]: I think it can stay as it is imported from grafana/ui

Comment (alexanderzobnin) [reply public/app/features/api-keys/ApiKeysPage.tsx]: @torkelo what do you think?

Comment (torkelo) [reply public/app/features/api-keys/ApiKeysPage.tsx]: Yea, could be nice with a higher level reusable component

**Review by torkelo — APPROVED** [review]
Looks good!

File: packages/grafana-ui/src/components/ConfirmButton/ConfirmButton.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            {children}
+          </Forms.Button>
+        ) : (
+          <span className={buttonClass} onClick={onClick}>
+            {children}
+          </span>
+        )}
+        <span className={styles.confirmButtonContainer}>
+          <span className={confirmButtonClass}>
+            <Button size={size} variant="transparent" onClick={this.onClickCancel}>
```
Comment (dprokop) [inline packages/grafana-ui/src/components/ConfirmButton/ConfirmButton.tsx]: Why not using Forms.Button?

**Review by dprokop — APPROVED** [review]
👍

File: packages/grafana-ui/src/components/ConfirmButton/ConfirmButton.tsx (hunk 3 above)
Comment (alexanderzobnin) [reply packages/grafana-ui/src/components/ConfirmButton/ConfirmButton.tsx]: `Forms.Button` has a blue border in a focused state, not sure it looks good in this context.

Comment (dprokop) [reply packages/grafana-ui/src/components/ConfirmButton/ConfirmButton.tsx]: That's by design/accessibility requirements. I think we should strive to use the new components if possible. Or maybe think about gentler way of indicating focus. @sarlinska @jessover9000 any thought on that?

**Pull request merged**